### PR TITLE
add `classifier_dropout` to classification heads

### PR DESCRIPTION
**Pull request by PhilipMay**
PR for #12781 and maybe a fix for #12792

# Logbook
- added to Electra: `SequenceClassification` and `TokenClassification`
- added to BERT: `SequenceClassification` and `TokenClassification`
- added to RoBERTa: `SequenceClassification` and `TokenClassification`
- added to big_bird: `SequenceClassification` and `TokenClassification`
- added to mobilebert: `SequenceClassification` and `TokenClassification`
- added to reformer: `SequenceClassification` <s>and `TokenClassification`</s>
- added to ConvBERT: `SequenceClassification` and `TokenClassification`
- added to albert: `SequenceClassification` and `TokenClassification`

### CODE REVIEW

File: src/transformers/models/electra/configuration_electra.py [hunk 1]
```diff
@@ -141,6 +145,7 @@ def __init__(
         summary_last_dropout=0.1,
         pad_token_id=0,
         position_embedding_type="absolute",
+        classifier_dropout: Optional[float] = None,
```
Comment (sgugger) [inline src/transformers/models/electra/configuration_electra.py:148]:
```suggestion
        classifier_dropout= None,
```
Let's not add type annotations just here. We can add them everywhere, but this should go in a separate PR.

File: src/transformers/models/electra/configuration_electra.py [hunk 2]
```diff
@@ -104,6 +106,8 @@ class ElectraConfig(PretrainedConfig):
             <https://arxiv.org/abs/1803.02155>`__. For more information on :obj:`"relative_key_query"`, please refer to
             `Method 4` in `Improve Transformer Models with Better Relative Position Embeddings (Huang et al.)
             <https://arxiv.org/abs/2009.13658>`__.
+        classifier_dropout:
+            The dropout ratio for classifier.
```
Comment (sgugger) [inline src/transformers/models/electra/configuration_electra.py:110]: This needs a bit more, see the other arguments:
```suggestion
        classifier_dropout (:obj:`float`, `optional`):
            The dropout ratio for the classification head.
```

**Review by sgugger — APPROVED** [review]
This can be added to the other models too!

**Comment by PhilipMay** [comment]
@sgugger should `classifier_dropout` be only added for `<model_name>ForSequenceClassification`
or also to `<model_name>ForTokenClassification`?

**Comment by sgugger** [comment]
It should be used for both, yes.

**Review by LysandreJik — APPROVED** [review]
LGTM

**Review by patrickvonplaten — APPROVED** [review]

**Comment by PhilipMay** [comment]
This is still WIP - I want to add more models. Please do not merge yet ...

**Comment by PhilipMay** [comment]
I am done with this PR. It is ready for review.
See logbook above for changed model types.

File: src/transformers/models/albert/modeling_albert.py [hunk 3]
```diff
@@ -1088,7 +1088,12 @@ def __init__(self, config):
         self.num_labels = config.num_labels
 
         self.albert = AlbertModel(config, add_pooling_layer=False)
-        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        classifier_dropout_prob = (
+            config.classifier_dropout_prob
+            if config.classifier_dropout_prob is not None
+            else config.hidden_dropout_prob
```
Comment (JetRunner) [inline src/transformers/models/albert/modeling_albert.py:1094]: I am against this behavior - it's a breaking change and it will create inconsistency and hurt the reproducibility of previous research code since the same config will give different results. The expected behavior is to set it as `0` when no `config.classifier_dropout_prob` is provided.

**Review by JetRunner — CHANGES_REQUESTED** [review]

**Review by JetRunner — APPROVED** [review]
Never mind :)

**Review by sgugger — APPROVED** [review]
Thanks again for working on this!
